frame/support/src/storage/mod.rs

```diff
@@ -557,6 +557,12 @@ pub trait StorageDoubleMap<K1: FullEncode, K2: FullEncode, V: FullCodec> {
 	where
 		KArg1: ?Sized + EncodeLike<K1>;
 
+	/// Does any value under the first key `k1` (explicitly) exist in storage?
+	/// Might have unexpected behaviour with empty keys, e.g. `[]`.
```

Comment:
Im wondering if we could mitigate this edge-case? Something similar to `key_slice.is_empty() return true`?

Reply:
I think empty keys don't make any sense in any kind of StorageMap. 
For the tests I have done, they can only be created using [u8; 0] as a type.

Reply:
I dont know how you would arrive at an empty key, since they are hash outputs.  
You can probably remove that comment to not confuse the reader.

Reply:
You could use Identity and force the keys to be empty, like this:
```rust
#[crate::storage_alias]
type FooDoubleMapWithEmptyArrays = StorageDoubleMap<Prefix, Identity, [u8; 0], Identity, [u8; 0], [u8; 0]>;
```
If the second key is empty and Identity, then the test fails.
```rust
assert_eq!(FooDoubleMapWithEmptyArrays::contains_prefix::<[u8; 0]>([]), false);
FooDoubleMapWithEmptyArrays::insert::<[u8; 0], [u8; 0], [u8; 0]>([], [], []);
assert!(FooDoubleMapWithEmptyArrays::contains_prefix::<[u8; 0]>([]));
FooDoubleMapWithEmptyArrays::remove::<[u8; 0], [u8; 0]>([], []);
assert_eq!(FooDoubleMapWithEmptyArrays::contains_prefix::<[u8; 0]>([]), false);
```

Reply:
Right… that would really be an edge-case. Then the doc is precise enough IMHO :+1: